com.ibm.streamsx.topology/com.ibm.streamsx.topology.functional.python/CCWindow/CCWindow_cpp.cgt

```diff
@@ -0,0 +1,253 @@
+/* Additional includes go here */
+
+#include "splpy.h"
+#include "splpy_funcop.h"
+
+using namespace streamsx::topology;
+
+<%SPL::CodeGen::implementationPrologue($model);%>
+
+@include "../pyspltuple.cgt"
+<%
+ # Select the Python wrapper function
+ my $pyoutstyle = splpy_tuplestyle($model->getOutputPortAt(0));
+ 
+ # Temporary, until more input types are supported. At the moment, only pass by reference types 

```

Comment:
The code is somewhat confusing as comments indicate it only supports pass by reference, but there is code to support dict and pickled values. Maybe the code could be cleaned up to only include what is currently supported?